src/finn/custom_op/fpgadataflow/thresholding_binary_search.py

```diff
@@ -0,0 +1,752 @@
+            in_ind += 1
+
+        # Create a PyVerilator wrapper of the RTLSim .so
+        sim = self.get_rtlsim()
+        nbits = self.get_instream_width()
+        inp = npy_to_rtlsim_input(
+            "{}/input_0.npy".format(code_gen_dir), export_idt, nbits
+        )
+
+        super().reset_rtlsim(sim)
+        super().toggle_clk(sim)
+
+        wnbits = self.get_weightstream_width()
+        export_wdt = self.get_weight_datatype()
+        wei = npy_to_rtlsim_input(
+            "{}/thresholds.npy".format(code_gen_dir), export_wdt, wnbits
+        )
+        num_w_reps = np.prod(self.get_nodeattr("numInputVectors"))
+        io_dict = {
+            "inputs": {"in0": inp, "weights": wei * num_w_reps},
+            "outputs": {"s_axis": []},
+        }
+        self.rtlsim_multi_io(sim, io_dict)
+        output = io_dict["outputs"]["out"]
+
+        # Manage output data
+        odt = self.get_output_datatype()
+        target_bits = odt.bitwidth()
+        packed_bits = self.get_outstream_width()
+        out_npy_path = "{}/output.npy".format(code_gen_dir)
+        out_shape = self.get_folded_output_shape()
+
+        rtlsim_output_to_npy(
+            output, out_npy_path, odt, out_shape, packed_bits, target_bits
+        )
+
+        # load and reshape output
+        output = np.load(out_npy_path)
+        oshape = self.get_normal_output_shape()
+        output = np.asarray([output], dtype=np.float32).reshape(*oshape)
+        context[node.output[0]] = output
+        return
+
+    def code_generation_ipi(self):
+        """Constructs and returns the TCL commands for node instantiation as an RTL
+        block."""
+        cmd = []
+        rtl_file_list = self.get_rtl_file_list()
+        code_gen_dir = self.get_nodeattr("code_gen_dir_ipgen")
+
+        for rtl_file in rtl_file_list:
+            cmd.append(
+                "add_files -norecurse %s" % (os.path.join(code_gen_dir, rtl_file))
+            )
+
+        # Create an RTL block, not an IP core (-type ip)
+        cmd.append(
+            "create_bd_cell -type module -reference %s %s"
+            % (self.get_nodeattr("gen_top_module"), self.onnx_node.name)
+        )
+
+        # ERROR: [BD 41-237] Bus Interface property FREQ_HZ does not match between
+        # /Thresholding_Binary_Search_0/s_axis(100000000 and
+        # /StreamingFIFO_0/out_V(200000000.000000)
+        cmd.append(
+            "set_property -dict [list CONFIG.FREQ_HZ {200000000}] [%s %s]"
+            % ("get_bd_intf_pins", "Thresholding_Binary_Search_0/s_axis")
+        )
+
+        # ERROR: [BD 41-237] Bus Interface property FREQ_HZ does not match between
+        # /StreamingFIFO_1/in0_V(200000000.000000) and
+        # /Thresholding_Binary_Search_0/m_axis(100000000)
+        cmd.append(
+            "set_property -dict [list CONFIG.FREQ_HZ {200000000}] [%s %s]"
+            % ("get_bd_intf_pins", "Thresholding_Binary_Search_0/m_axis")
+        )
```

Comment:
why is this property being explicitly set here? the errors in the comments could indeed be raised because of this

Reply:
added attribute settings here: https://github.com/Xilinx/finn/pull/687/commits/e2816d3e1c8ce75ad9f0b1aafbef25af8b305a6c